keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,491 @@
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently the most container runtimes work is by masking and setting as read-only certain paths in `/proc`.
+This is to prevent data from being exposed into a container that should not be.
+However, there are certain use-cases where it is necessary to turn this off.
+
+In 1.12, this was introduced as the ProcMountType feature gate, and has languished since. This KEP is
+a successor to (and heavily based on) https://github.com/kubernetes/community/pull/1934/, updated for the modern era.
+
+## Motivation
+
+For end-users who would like to run unprivileged containers using user namespaces _nested inside_ CRI containers, we need an option to have a `ProcMount`.
+That is, we need an option to designate explicitly turn off masking and setting read-only of paths so that we can mount `/proc` in the nested container as an unprivileged user.
+
+Please see the following filed issues for more information:
+- [opencontainers/runc#1658](https://github.com/opencontainers/runc/issues/1658#issuecomment-373122073)
+- [moby/moby#36597](https://github.com/moby/moby/issues/36597)
+- [moby/moby#36644](https://github.com/moby/moby/pull/36644)
+
+### Goals
+
+- Add the ProcMountType option to the pod API, allowing users to override the proc mount behavior in a container.
+
+### Non-Goals
+
+## Proposal
+
+
+Add a new `string` type field named `ProcMountType` will hold the viable
+options for `procMount` to the `SecurityContext`
+definition.
+
+By default,`procMount` is `default`, aka the same behavior as today and the
+paths are masked.
```

Comment:
should we have Masked as an explicit option? If this wasn't a part of an API before, making an explicit Masked option would be a better design

Reply:
it will ensure that the user may force masking as a security feature

Reply:
I am less inclined personally--feels like syntactic sugar. Functionally, default is masked, and a use could just as easily enforce default (see PSS today) as masked

Reply:
discussed with @mrunalp - it may be a good idea to change the descriprion of "default" to say it means that certain subpaths are masked. Link to or oci list https://github.com/kubernetes/kubernetes/blob/598076a71caeaf7720b758eedd5cca5a6364ef62/pkg/securitycontext/util.go#L193-L211

If default meaning is "whatever", I wonder if we can even write an e2e test validating any behavior for the "default"

Reply:
+1 to clarifying the definition: would you like me to capture that work here, and how? 

We can test what default means because it's interpreted by the kubelet, so it's all here. A better situation than if we passed the proc mount type to CRI and then it decided what to do with the field. We can test for the masked value of the files in the container's rootfs.

Reply:
yes, let's mention that we will clarify the semantic of this field

Reply:
see comment ^ IMO default allows the container runtime to configure .. much as we handle other default isolation tooling options..